modules/basics/src/main/java/com/opengamma/basics/date/ImmutableHolidayCalendar.java

```diff
@@ -141,11 +156,33 @@ private ImmutableHolidayCalendar(String name, SortedSet<LocalDate> holidays, Set
     this.holidays = ImmutableSortedSet.copyOfSorted(holidays);
     this.weekendDays = Sets.immutableEnumSet(weekendDays);
     if (holidays.isEmpty()) {
-      range = LocalDateRange.ALL;
+      this.range = LocalDateRange.ALL;
+      this.startYear = 0;
+      this.lookup = new int[0];
     } else {
       this.range = LocalDateRange.ofClosed(
           holidays.first().with(TemporalAdjusters.firstDayOfYear()),
           holidays.last().with(TemporalAdjusters.lastDayOfYear()));
+      this.startYear = range.getStart().getYear();
+      int[] array = new int[(range.getEndExclusive().getYear() - startYear) * 12];
```

Comment:
I can see that this is correct but just wonder if it could be clearer: the range is set so that the last included date is the last day of the year, so`` range.getEndExclusive()`` must be in the next year but it's easy to overlook the inclusive/exclusive difference. Maybe if you put the last day in a var and then used that directly in the size calculation (with a +1 of course), it might be clearer.